pkg/sql/row/row_converter.go

```diff
@@ -264,29 +264,25 @@ func (c *DatumRowConverter) getSequenceAnnotation(
 
 	var seqNameToMetadata map[string]*SequenceMetadata
 	var seqIDToMetadata map[descpb.ID]*SequenceMetadata
+	cf := descs.NewBareBonesCollectionFactory(evalCtx.Settings, evalCtx.Codec)
+	descsCol := cf.MakeCollection(descs.NewTemporarySchemaProvider(evalCtx.SessionDataStack))
```

Comment:
there's something upsetting about this whole dance. We can leave it for now, but it's very crufty.